lib/xdrgen/generators/go.rb

```diff
@@ -389,7 +389,16 @@ def render_typedef_encode_to_interface(out, typedef)
         name = name(typedef)
         type = typedef.declaration.type
         out.puts "// EncodeTo encodes this value using the Encoder."
-        out.puts "func (s #{name}) EncodeTo(e *xdr.Encoder) error {"
+
+        if type.is_a?(AST::Typespecs::Opaque) and type.fixed?
+          # Implement EncodeTo by pointer in fixed opaque types (arrays)
+          # otherwise (if called by value), Go will make a heap allocation
```

Comment:
💡 Suggested rewording of this explanation:
```suggestion
          # Make the receiver of EncodeTo a pointer in fixed opaque types (arrays)
          # otherwise (if called by value), Go will make a heap allocation
```